### ISSUE TITLE
call information

### ISSUE THREAD

**Issue by Savaturkey** [chenxiaolong/BCR#3]
Your program is beautiful. It will be very nice to find the caller information, timestamp, who called and incoming or outgoing call information on the file name where the call is saved.

**Comment by chenxiaolong** [comment]
Thanks!

The filename should already include the timestamp and phone number. The format currently looks like this:
```
<date>_<time><timezone>_<phone number>.flac
```
Eg.
```
20220522_122802.881926-0400_+12345678900.flac
```

I'll see if I can add `in` or `out` to the filename to distinguish between incoming and outgoing calls.

**Comment by Savaturkey** [comment]
The name of the caller or the person being called is also very important. I don't think the time frame matters.

**Comment by chenxiaolong** [comment]
I implemented some of the suggestions in #11, #12, and #13.

* The filenames now include `in` or `out` on Android 10+. The API does not exist on Android 9.
* Incoming calls will now include the caller name in the filename. This is based on the CNAP value (caller ID) at the time that the call is answered. If you have a third party caller ID app that's slow and updates after the call is answered, the filename will **not** reflect the updated caller ID.

  Also, BCR will **not** look up names in the contact database. I currently do not plan on implementing any features that require the contacts permission.
* The timestamps in the filenames now match the dialer's call log exactly. It represents the time that Android registered the incoming/outgoing call instead of the time that the recording began.

The output formats are now:
```
# Incoming calls (Android 10+)
<date>_<time><timezone offset>_in_<phone number>_<caller ID>.flac

# Incoming calls (Android 9)
<date>_<time><timezone offset>_<phone number>_<caller ID>.flac

# Outgoing calls (Android 10+)
<date>_<time><timezone offset>_out_<phone number>.flac

# Outgoing calls (Android 9)
<date>_<time><timezone offset>_<phone number>.flac
```

(I'm keeping the timezone offset in the filename because it's the only way to distinguish timestamps across DST boundaries when the time jumps backwards.)

**Comment by DarkKnight1995** [comment]
> I implemented some of the suggestions in #11, #12, and #13.
> 
> * The filenames now include `in` or `out` on Android 10+. The API does not exist on Android 9.
> * Incoming calls will now include the caller name in the filename. This is based on the CNAP value (caller ID) at the time that the call is answered. If you have a third party caller ID app that's slow and updates after the call is answered, the filename will **not** reflect the updated caller ID.
>   Also, BCR will **not** look up names in the contact database. I currently do not plan on implementing any features that require the contacts permission.
> * The timestamps in the filenames now match the dialer's call log exactly. It represents the time that Android registered the incoming/outgoing call instead of the time that the recording began.
> 
> The output formats are now:
> 
> ```
> # Incoming calls (Android 10+)
> <date>_<time><timezone offset>_in_<phone number>_<caller ID>.flac
> 
> # Incoming calls (Android 9)
> <date>_<time><timezone offset>_<phone number>_<caller ID>.flac
> 
> # Outgoing calls (Android 10+)
> <date>_<time><timezone offset>_out_<phone number>.flac
> 
> # Outgoing calls (Android 9)
> <date>_<time><timezone offset>_<phone number>.flac
> ```
> 
> (I'm keeping the timezone offset in the filename because it's the only way to distinguish timestamps across DST boundaries when the time jumps backwards.)

For android 9, may be rearrange the naming sequence for in/out differentiation, for example, for incoming calls name the file using the current scheme and for outgoing calls, bring the number to the front of the filename like;

<phone number>_<date>_<time><timezone offset>.flac
